rpcs3/Emu/RSX/Overlays/overlay_message.cpp

```diff
@@ -0,0 +1,108 @@
+#include "stdafx.h"
+#include "overlay_message.h"
+#include "Emu/RSX/RSXThread.h"
+
+namespace rsx
+{
+	namespace overlays
+	{
+		message_item::message_item(localized_string_id msg_id)
+		{
+			m_expiration_time = get_system_time() + 5'000'000;
+
+			m_text.set_font("Arial", 16);
+			m_text.set_text(msg_id);
+			m_text.auto_resize();
+			m_text.back_color.a = 0.f;
+
+			m_fade_animation.current = color4f(1.f);
+			m_fade_animation.end = color4f(1.0f, 1.0f, 1.0f, 0.f);
+			m_fade_animation.duration = 2.f;
+			m_fade_animation.active = true;
+		}
+
+		u64 message_item::get_expiration() const
+		{
+			return m_expiration_time;
+		}
+
+		compiled_resource message_item::get_compiled()
+		{
+			if (!m_processed)
+			{
+				return {};
+			}
+
+			auto cr = m_text.get_compiled();
+			m_fade_animation.apply(cr);
+
+			return cr;
+		}
+
+		void message_item::update(usz index, u64 time)
+		{
+			if (m_cur_pos != index)
+			{
+				m_cur_pos = index;
+				m_text.set_pos(10, index * 18);
+			}
+
+			if ((m_expiration_time - time) < 2'000'000)
```

Comment:
```suggestion
			if (fade_animation.active)
```

Reply:
I only want to apply it when there is less than 2 seconds left...